pytorch_lightning/trainer/trainer.py

```diff
@@ -420,7 +420,7 @@ def fit(
         # we reuse fit for other functions. When already set, it shouldn't be modified.
         if not self.state.running:
             self.state = TrainerState.FITTING
-        if self._running_stage is None:
+        if self._running_stage is None or self._running_stage == RunningStage.TUNING:
```

Comment:
this is for test failure `trainer.tune(model)` here
https://github.com/PyTorchLightning/pytorch-lightning/blob/master/tests/trainer/test_states.py#L45-L48

Reply:
```suggestion
        if self._running_stage is None or self.tuning:
```

The tuning case is different because it is treated as training internally, we might want to set the stage back to tuning at the end of this, if tuning was initially set